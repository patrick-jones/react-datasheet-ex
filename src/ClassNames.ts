export const READ_ONLY_CELL = 'cell read-only';
export const ACTION_CELL = 'rdx-action-cell';

export const HEADER_CELL = 'rdx-header-cell';
export const HEADER_CELL_LEFT = 'rdx-header-cell--left';
export const HEADER_CELL_CENTER = 'rdx-header-cell--center';
export const HEADER_CELL_RIGHT = 'rdx-header-cell--right';

export const HEADER_CELL_CONTAINER = 'rdx-header-cell-container';
export const HEADER_CELL_TITLE = 'rdx-header-cell-title';

export const HEADER_DRAG_SOURCE = 'rdx-header-drag-source';
export const HEADER_DROP_TARGET = 'rdx-header-drop-target';

export const ROW = 'rdx-row';
export const ROW_DROP_TARGET = 'rdx-row-drop-target';
export const ROW_DRAG_SOURCE = 'rdx-row-drag-source';

export const RESIZING_WIDTH = 'rdx-ex-resizing-width';
export const DATA_GRID_RESIZABLE = 'rdx-data-grid-resizable';
export const WIDTH_RESIZE_HANDLE = 'rdx-width-resize-handle';

export const SCROLL_PANEL = 'rds-ex-scroll-panel';
