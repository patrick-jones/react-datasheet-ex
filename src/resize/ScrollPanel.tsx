import React, {PureComponent, ReactNode} from 'react';
import {SCROLL_PANEL} from '../ClassNames';


export interface ScrollPanelProps {
  width?: number;
  height?: number;
}


export default class ScrollPanel
  extends PureComponent<ScrollPanelProps> {

  render(): ReactNode {
    const {width, height} = this.props;
    const style = {
      width: width ? `${width}px` : undefined,
      height: height ? `${height}px` : undefined,
    };
    return (
      <span className={SCROLL_PANEL} style={style}>
        {this.props.children}
      </span>
    );
  }
}
