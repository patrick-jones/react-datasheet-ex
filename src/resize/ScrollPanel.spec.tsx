import React from 'react';
import 'jest';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import ScrollPanel from './ScrollPanel';
import {SCROLL_PANEL} from '../ClassNames';

describe('ScrollPanel', () => {
  it('Should render', () => {
    const wrapper = shallow(<ScrollPanel/>);
    expect(wrapper.prop('style')).toEqual({});
    expect(wrapper.hasClass(SCROLL_PANEL)).toBe(true);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should render optional styles', () => {
    const wrapper = shallow(<ScrollPanel height={200} width={450} />);
    expect(wrapper.prop('style')).toEqual({height: '200px', width: '450px'});
    expect(wrapper.hasClass(SCROLL_PANEL)).toBe(true);
  });
});
