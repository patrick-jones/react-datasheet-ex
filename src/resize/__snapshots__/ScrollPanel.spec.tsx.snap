// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollPanel Should render 1`] = `
<span
  className="rds-ex-scroll-panel"
  style={
    Object {
      "height": undefined,
      "width": undefined,
    }
  }
/>
`;
